retry/retry.go

```diff
@@ -247,7 +247,11 @@ func (r *retrier) Next() bool {
 func (r retrier) retryIn() time.Duration {
 	backoff := float64(r.options.initialBackoff) * math.Pow(r.options.multiplier, float64(r.currentAttempt))
 	if r.options.maxBackoff != 0 && backoff > float64(r.options.maxBackoff) {
-		backoff = float64(r.options.maxBackoff)
+		if r.options.initialBackoff > r.options.maxBackoff {
```

Comment:
Would it be better to set `maxBackoff = max(maxBackoff, initialBackoff)` in `Start()` after applying the params so we don't do this check every time, given that the logic would be the same? I might consider this semantic more "construction-time" than "runtime"

Reply:
I considered that and ultimately chose not to do so because it mucks with what is currently a very clean/understandable `Start` with initialization and then application of option. I'm happy to put it there instead